services/QuillLMS/spec/models/concept_feedback_spec.rb

```diff
@@ -74,39 +75,116 @@
       it 'adds the translations to the data' do
         expect(subject["translatedDescription"]).to eq(translation)
       end
+
+      context 'there is only a gengo translation' do
+        let(:translated_text) { create(:gengo_translated_text, translation: translation)}
+
+        it 'adds the translations to the data' do
+          expect(subject["translatedDescription"]).to eq(translation)
+        end
+      end
+
+      context 'there is a gengo and open ai translation' do
+        let(:gengo_text) { create(:gengo_translated_text)}
+
+        before do
+          concept_feedback.english_texts.first.translated_texts = [gengo_text, translated_text]
+        end
+
+        it 'defaults to the open_ai translation' do
+          expect(subject["translatedDescription"]).to eq(translation)
+        end
+
+        context 'passed in gengo as an option' do
+          let(:options) { {source_api: TranslatedText::GENGO_SOURCE} }
+
+          it 'returns the gengo translation' do
+            expect(subject["translatedDescription"]).to eq(gengo_text.translation)
+          end
+        end
+
+      end
     end
   end
 
-  describe '#translation(locale:)' do
+  describe '#translation(locale:, source:)' do
     let(:locale) {"es-la"}
+    let(:translation_locale) { locale }
+    let(:source) { TranslatedText::OPEN_AI_SOURCE }
+
+    context 'source is not passed in' do
+      subject { concept_feedback.translation(locale: locale)}
+
+      context 'a translation exists' do
+        let(:translation) { "test translation" }
+        let(:translated_text) do
+          create(:translated_text,
+           translation: translation,
+           locale: translation_locale,
```

Comment:
I think we can just use the keys here if we use `locale` instead of `translation_locale`

Reply:
this one is so that it's overwritable so we do need two diff names